### PR TITLE
Add extra attribute to `AccessibleButton`

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

Part of https://github.com/element-hq/element-web/issues/26762

Prepare work for https://github.com/matrix-org/matrix-react-sdk/pull/12448 (it will be splitted in many PRs)
I will need this extra attributes for `AccessibleButton` for the migration from `AccessibleTooltipButton`.

TLDR: Add new attribute to prepare the incoming migration to  from `AccessibleTooltipButton` to `AccessibleButton`

Also, the `triggerInteractive` creates unwanted extra span in the dom.

### CODE REVIEW

File: src/components/views/elements/AccessibleButton.tsx [hunk 1]
```diff
@@ -96,6 +98,14 @@ type Props<T extends keyof JSX.IntrinsicElements> = DynamicHtmlElementProps<T> &
...
      */
     caption?: string;
+    /**
+     * The placement of the tooltip.
+     */
+    placement?: TooltipProps["placement"];
+    /**
+     * Callback for when the tooltip is opened or closed.
+     */
+    onOpenChange?: TooltipProps["onOpenChange"];
```
Comment (t3chguy) [inline src/components/views/elements/AccessibleButton.tsx:108]: onTooltipOpenChange maybe?
